### PR TITLE
Send events as native macOS notifications

### PR DESCRIPTION
**Pull request by lgalabru**
This PR is about making electron a good inter-process-communication citizen.
I'm working on an electron app, using a native helper/agent/daemon as a backend. 
I've been handling IPC using sockets. It works, but it's heavy, and not a great practice. 
Since electron supports receiving distributed notifications, it would be great to also make it able to send some of them.

### CODE REVIEW

File: atom/browser/api/atom_api_system_preferences_mac.mm [hunk 1]
```diff
@@ -28,6 +28,28 @@
...
+    const base::DictionaryValue& user_info) {
+  DoPostNotification(name, user_info, false);
+}
+
+void SystemPreferences::PostLocalNotification(const std::string& name, 
+    const base::DictionaryValue& user_info) {
+  DoPostNotification(name, user_info, true);
+}
+
+void SystemPreferences::DoPostNotification(const std::string& name,
```
Comment (miniak) [inline atom/browser/api/atom_api_system_preferences_mac.mm]: Do you need this helper method, if the implementation is different depending on the is_local flag?

File: atom/browser/api/atom_api_system_preferences_mac.mm [hunk 2]
```diff
@@ -28,6 +28,28 @@
...
+  DoPostNotification(name, user_info, true);
+}
+
+void SystemPreferences::DoPostNotification(const std::string& name, 
+    const base::DictionaryValue& user_info, bool is_local) {
+  NSNotificationCenter* center = is_local ?
+    [NSNotificationCenter defaultCenter] :
+    [NSDistributedNotificationCenter defaultCenter];
+  [center
+     postNotificationName:base::SysUTF8ToNSString(name)
```
Comment (miniak) [inline atom/browser/api/atom_api_system_preferences_mac.mm]: Can you just use `base::SysUTF8ToNSString`?

File: atom/browser/api/atom_api_system_preferences_mac.mm [hunk 3]
```diff
@@ -28,6 +28,28 @@
...
+    const base::DictionaryValue& user_info) {
+  DoPostNotification(name, user_info, true);
+}
+
+void SystemPreferences::DoPostNotification(const std::string& name, 
+    const base::DictionaryValue& user_info, bool is_local) {
+  NSNotificationCenter* center = is_local ?
+    [NSNotificationCenter defaultCenter] :
+    [NSDistributedNotificationCenter defaultCenter];
+  [center
```
Comment (miniak) [inline atom/browser/api/atom_api_system_preferences_mac.mm]: The colons should be aligned, like this:

```
[center
  postNotificationName:base::SysUTF8ToNSString(name)
                object:nil
              userInfo:DictionaryValueToNSDictionary(user_info)
];
```

File: atom/browser/api/atom_api_system_preferences.h [hunk 4]
```diff
@@ -9,6 +9,7 @@
 
 #include "atom/browser/api/event_emitter.h"
 #include "base/callback.h"
+#include "base/values.h"
```
Comment (zcbenz) [inline atom/browser/api/atom_api_system_preferences.h]: This line is redundant, necessary types should have already been declared.

Comment (lgalabru) [inline atom/browser/api/atom_api_system_preferences.h]: That's what I though, but for some reason, the CI is failing if we don't include this header :/
https://travis-ci.org/electron/electron/jobs/150172395

**Comment by zcbenz** [comment]
👍

**Pull request merged**